freeze_room/__init__.py

```diff
@@ -310,7 +292,38 @@ async def _on_room_leave(
             }
         )
 
-        return True
+        return
+
+    async def _promote_to_admins(
+        self,
+        users_to_promote: Iterable[str],
+        pl_content: dict,
+        event: EventBase,
+    ) -> None:
+        """Promotes a given list of users to admins.
+
+        Args:
+            users_to_promote: The users to promote.
+            pl_content: The content of the m.room.power_levels event that's currently in
+                the room state.
+            event: The event we want to use the sender and room_id of to send the new
+                power levels event.
+        """
+        # Make a deep copy of the content so we don't edit the "users" dict from
+        # the event that's currently in the room's state.
+        new_pl_content = copy.deepcopy(pl_content)
+        for user in users_to_promote:
+            new_pl_content["users"][user] = 100
```

Comment:
It is not, strictly speaking, a valid assumption that 100 means "admin". Should we instead set it to whatever PL `sender` currently has?

Reply:
Good point.